### ISSUE TITLE
[Headeritems][Accessibility]:Incorrect control type is given for the "Name, Created at, Created By, Status" columns headers.

### ISSUE THREAD

**Issue by Charanrajkb** [OfficeDev/office-ui-fabric-react#11977]
**Environment Details:​​**
Browser Versions: Microsoft Edge 44.18362.449.0
Chrome Version : Version 79.0.3945.88 (Official Build) (64-bit)
Windows Version: Windows10​​

**Repro Steps:​​**
1. Hit the application  URL
2. Select application as "Sales hub" from the list of application.​
3. From the change area which is present in the bottom select "App Settings"​
4. Navigate to "Site Map" and then Navigate to "Goal management" and select "Forecast Configuration" ​
5. Navigate to grid and select "More Actions" button from the data cells under the column Forecast name and hit enter.​
6. Select Add/View Snapshot menu item and press enter.​
7. "Snapshot History" screen will appears.​
8. Open Accessibility Insights for desktop and check the name of the table headers.

**Actual Results:**
Incorrect control type is given for the "Name, Created at, Created By, Status" columns headers.
The control is given as button which is incorrect.

**Expected:​​**
Control type of the "Name, Created at, Created By, Status" columns headers must be header item and not as buttons.

**User Impact:​**
Screen reader users will face difficulty to understand whether column headers are buttons which are actionable or column headers. 
Screen readers will not understand the table headers are present for the table.

Attachment:
![1677497_A11y_D365DealflowPredictivechartingfeatures_Snapshot History_AI4D (1)](https://user-images.githubusercontent.com/52071743/73823659-bd473280-481e-11ea-8198-063ddf22363e.png)

**Comment by Charanrajkb** [comment]
Please find the details below:
1)Which Fabric component and which version of the Fabric component is causing the issue:
The header panel in the table is causing the issue.

2)What behaviors and attributes are missing or incorrect:
The control type is incorrect for the headers item.

3)What you expected and what is actually happening:
The control type must be header item not as button.

4)Confirmation that the problem reproduces in isolation:Yes

**Comment by xugao** [comment]
@Charanrajkb - could you pls provide a codepen or link to our example page to demo the issue?

here is what i see with [this example list](https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/basic):
![image](https://user-images.githubusercontent.com/1207059/74861788-c98fbd00-52ff-11ea-9aa9-47c4dac7aaf7.png)

**Comment by Charanrajkb** [comment]
Please find the screenshot below:
![headerItem](https://user-images.githubusercontent.com/52071743/74904344-a7ff0600-53d1-11ea-8c16-1ec353ae9a1d.png)

**Comment by xugao** [comment]
@chakkaradeep - it is by design a column header containing a button (click to select an option to sort)
![image](https://user-images.githubusercontent.com/1207059/74964414-41c1b580-53c8-11ea-957b-dbf3c170d67b.png)

I don't see a bug here. Pls point out if you otherwise and why.
Or close issue if this answers you :)

**Comment by Charanrajkb** [comment]
User Impact:
Screen reader users will find very difficult in differentiating between button and the table header items.
When the focus is on the table header items there is no action is performed after activating them.So it is good to give the control type as header items for the table headers and not as the buttons. 

Attachment:
[11977_[Headeritems][Accessibility].pptx](https://github.com/OfficeDev/office-ui-fabric-react/files/4248120/11977_.Headeritems.Accessibility.pptx)

**Comment by Charanrajkb** [comment]
@xugao Please find the screenshot and the screen recording.
https://user-images.githubusercontent.com/52071743/74904468-eb597480-53d1-11ea-9c03-ca8f28db9330.png

Attachment:
[11977_.Headeritems.Accessibility.pptx](https://github.com/OfficeDev/office-ui-fabric-react/files/4266431/11977_.Headeritems.Accessibility.pptx)

**Comment by xugao** [comment]
@Charanrajkb - the issue you having is likely due to your incorrect implementation (using the component interface incorrectly). like i said before, see [this example](https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/basic), it does not contain button. This is why I am asking for code which demos your implementation so i can advice you on what the correct implementation is.

**Comment by Charanrajkb** [comment]
@xugao please find the copdeopen:
![image](https://user-images.githubusercontent.com/52071743/75605505-7db3f380-5b09-11ea-8877-43a03da73639.png)

The role must not be button and the role must be header item.If the role is given as header item keyboard focus will not move to the table headers.

**Comment by xugao** [comment]
@Charanrajkb - the screenshot doesn't demo the code..

can you pls check out our example and mimic how this is being implemented:
https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/basic
![image](https://user-images.githubusercontent.com/1207059/75646970-3cead600-5c00-11ea-80d5-815afd841707.png)

**Comment by xugao** [comment]
@Charanrajkb - make sure in your implementation, do you set `onColumnHeaderClick` on `DetailsList`. if you do set `onColumnHeaderClick`, the header will become a role button

**Comment by xugao** [comment]
followed up offline. this is an issue with particular implementation, not the component itself.
